### ISSUE TITLE
BCTide mismatch between `nbfr` and line count

### ISSUE THREAD

**Issue by SorooshMani-NOAA** [schism-dev/pyschism#116]
This is my driver setup for writing a simple tidal run:
```python
    tidal_flags = [3, 3, 0, 0]
    config = ModelConfig(
        hgrid=hgrid,
        fgrid=fgrid,
        flags=[tidal_flags for _ in hgrid.boundaries.open.itertuples()],
        constituents='all',
        database='tpxo',
    )
```
And this is a section of the `bctides.in` that I get, which doesn't look right:

```
   19 8 !nbfr
   20 M2
   21  0.000140519 0.966301 294.637
   22 S2
   23  0.000145444 1 0
   24 N2
   25  0.00013788 0.966301 195.94
   26 K2
   27  0.000145842 1.29096 73.4175
   28 K1
   29  7.29212E-05 1.1055 306.893
   30 O1
   31  6.75977E-05 1.17112 349.005
   32 P1
   33  7.25229E-05 1 49.9675
   34 Q1
   35  6.49585E-05 1.17112 250.307
   36 Mm
   37  2.6392E-06 0.88214 98.6978
   38 Mf
   39  5.32341E-06 1.4179 136.627
   40 M4
   41  0.000281038 0.933738 229.275
   42 MN4
   43  0.000278399 0.933738 130.577
   44 MS4
   45  0.000285963 0.966301 294.637
   46 2N2
   47  0.00013524 0.966301 97.2419
   48 S1
   49  7.27221E-05 1 180
```

**Comment by SorooshMani-NOAA** [comment]
Sure, I'll take a look when I get a chance. Feel free to assign it to me (I don't have permission to assign to myself)

**Comment by SorooshMani-NOAA** [comment]
I just realized there is a fix by Linlin after the version I was using! I don't know how I missed it earlier. I'm sorry about the confusion. I'll close the ticket. This is the fix #110
